### PR TITLE
Update to Julia@v1.11 & CUDA.jl@v5.5.2

### PR DESCRIPTION
**Pull request by findmyway**
I also added a github action to upload the generated docker images. Feel free to adjust it as necessary.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,8 +10,8 @@ ARG JULIA_VERSION=1.9.0-rc3
 RUN apt-get update && \
     DEBIAN_FRONTEND=noninteractive \
     apt-get install --yes --no-install-recommends \
-                    # basic stuff
-                    curl ca-certificates nano && \
+    # basic stuff
+    curl ca-certificates nano && \
```
Comment (maleadt) [inline Dockerfile]: Misleading whitespace?

**Review by maleadt — CHANGES_REQUESTED** [review]
I think you need more changes than this, e.g., updating Julia in the Dockerfile to match the manifest, updating the preferences (version=local changed to local=true + version=...), etc.

File: Dockerfile (hunk 1 above)
Comment (findmyway) [reply Dockerfile]: Reverted

**Comment by findmyway** [comment]
> updating Julia in the Dockerfile to match the manifest

I'm very sure about this point. The Julia version is specified through docker build arguments. To make the `Manifest.toml` compatible with Julia version. I added an extra `Manifest-v1.11.toml` based on the doc here https://pkgdocs.julialang.org/v1/toml-files/#Different-Manifests-for-Different-Julia-versions

Feel free to suggest more improvements.

File: LocalPreferences.toml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 [CUDA_Runtime_jll]
-version = "local"
+__clear__ = ["version"]
```
Comment (maleadt) [inline LocalPreferences.toml:2]: Why the clear?

**Comment by maleadt** [comment]
Alternative, we could get rid of the Manifest altogether and simply use compat bounds. That would make the arg the single point of truth. I can take a look if you want (or feel free to experiment yourself).

File: LocalPreferences.toml (hunk 2 above)
Comment (findmyway) [reply LocalPreferences.toml:2]: I guess this is auto generated?

TBH, I'm not sure how to set the `version` here.

**Comment by maleadt** [comment]
I've opted to fold these improvements into the existing Dockerfile of the CUDA.jl repository: https://github.com/JuliaGPU/CUDA.jl/commit/6f9bcaa12b0047ac8295caa2599da2ba680e788b

You can now pull the `dev` tag from ghcr.io, and I'll built v5.5.2 too as the latest stable tag: https://github.com/JuliaGPU/CUDA.jl/pkgs/container/cuda.jl

**Pull request closed without merging**